src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -145,37 +91,14 @@ export const ChainMenu = memo((props: ChainMenuProps) => {
 
   return (
     <Box {...props}>
-      <Menu autoSelect={false}>
-        <Tooltip
-          label={translate(
-            currentChainNativeAsset ? 'common.switchNetwork' : 'common.unsupportedNetwork',
-          )}
-          isDisabled={!canSwitchChains}
-        >
-          <MenuButton as={Button} iconSpacing={2} px={2} width={menuButtonWidth}>
-            <Flex alignItems='center' justifyContent='center'>
-              {currentChainNativeAsset ? (
-                <AssetIcon assetId={currentChainNativeAssetId} showNetworkIcon size='xs' />
-              ) : (
-                <WarningIcon color='yellow.300' boxSize='4' />
-              )}
-            </Flex>
-          </MenuButton>
-        </Tooltip>
-
-        <MenuList p='10px' zIndex={2}>
-          <MenuGroup title={translate('common.selectNetwork')} ml={3} color='text.subtle'>
-            {supportedEvmChainIds.map(chainId => (
-              <ChainMenuItem
-                isConnected={chainId === connectedEvmChainId}
-                key={chainId}
-                chainId={chainId}
-                onClick={handleChainClick}
-              />
-            ))}
-          </MenuGroup>
-        </MenuList>
-      </Menu>
+      <GenericChainMenu<ChainId>
```

Comment:
If we can't avoid using a generic for component props, can we at least reassign this beforehand to avoid having this generic in-place? Components with generics at JSX render time feel super odd

Reply:
Also since it feels like `ChainId` will be the most common case, followed by `EvmChainId`, `UtxoChainId` and `'All'`, what about exporting those flavours directly from the `ChainMenu` module?

Reply:
ooh i like where your head is at, will get this done

Reply:
fcc2228dc6091e6a2e1afe42e981c72523e266d1